pkg/apis/networking/types.go

```diff
@@ -336,19 +336,56 @@ type HTTPIngressRuleValue struct {
 	// options usable by a loadbalancer, like http keep-alive.
 }
 
-// HTTPIngressPath associates a path regex with a backend. Incoming urls matching
-// the path are forwarded to the backend.
+// PathType represents the type of path referred to by a HTTPIngressPath.
+type PathType string
+
+const (
+	// PathTypeExact matches the URL path exactly.
+	PathTypeExact = PathType("Exact")
+
+	// PathTypePrefix matches based on a URL path prefix split by '/'. Matching is
+	// done on a path element by element basis. A path element refers is the
+	// list of labels in the path split by the '/' separator. A request is a
+	// match for path p if every p is an element-wise prefix of p of the
+	// request path. Note that if the last element of the path is a substring
+	// of the last element in request path, it is not a match (e.g. /foo/bar
```

Comment:
What's the rationale for this match rule?

Reply:
Just to clarify, is this question about why we're adding a Prefix path type, or the specifics of what we define as a prefix match in this comment?

Reply:
This comment can be resolved.  We discussed somewhere else.